provider/pkg/e2e_test.go

```diff
@@ -15,7 +15,41 @@ import (
 )
 
 func TestApi(t *testing.T) {
-	e2eTest("api").Run(t)
+	e2eTest("api", providertest.WithBaselineVersion("1.0.1")).Run(t)
+}
+
+func TestBinaryMediaTypesNoInput(t *testing.T) {
+	e2eTest("binaryMediaTypesNoInput", providertest.WithProviderName("aws-apigateway"),
```

Comment:
Ah this is interesting, looks like this is included upgrade tests in this PR although I suspect there's no particular concern about testing upgrades for this new feature, that's just overhead. CC @danielrbradley FYI per our recent conversation, it might be less confusing to separate out upgrade tests into their own `func Test*` roots.

Reply:
We specifically wanted to add upgrade tests since the first iteration of this change forced replaces in resources.

@pierskarsenbarg  came up with a solution after which doesn't require that but I we need to add some tests for the various configurations of this, in order to make sure that none of them are broken by the changes.

Reply:
Ah excellent. So that's intentional then. Thank you!